lib/main.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |x|
+      new_arr << x if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all(arg = nil)
+    unless block_given?
+      falsy = 0
+      my_each do |x|
+        falsy += 1 if x == false || x.nil?
+      end
+      return falsy.zero? if arg.nil?
+
+      return self == [arg]
+    end
+
+    pos = 0
+    my_each do |x|
+      pos += 1 if yield(x) == true
+    end
+    pos == size
+  end
+
+  def my_any(arg = nil)
```

Comment:
remember to use the correct name of the methods, for example, you are now trying to emulate [any?](https://apidock.com/ruby/Enumerable/any%3F)